src/main/java/org/folio/rest/util/CustomFieldsManager.java

```diff
@@ -0,0 +1,151 @@
+        .compose(o -> requestCustomFieldsDefinitions(httpClient, headers))
+        .compose(jsonObjects -> updateCF(jsonObjects, customFieldsOptions, httpClient, headers))
+        .setHandler(o -> {
+          httpClient.closeClient();
+          if (o.succeeded()) {
+            future.complete();
+          } else {
+            future.fail(o.cause());
+          }
+        });
+    }
+    return future;
+  }
+
+  @SuppressWarnings("unchecked")
+  private static Map<String, Set<String>> getCustomFieldsOptions(UserdataimportCollection userCollection) {
+    Map<String, Set<String>> customFieldsOptions = new HashMap<>();
+    List<CustomFields> customFields = userCollection.getUsers()
+      .stream()
+      .map(User::getCustomFields)
+      .filter(Objects::nonNull)
+      .collect(Collectors.toList());
+    for (CustomFields customField : customFields) {
+      customField.getAdditionalProperties().forEach((s, o) -> {
+        Set<String> options = customFieldsOptions.computeIfAbsent(s, s1 -> new HashSet<>());
+        if (o instanceof String) {
+          options.add((String) o);
+        } else if (o instanceof List) {
+          options.addAll((List<String>) o);
+        }
+      });
+    }
+    return customFieldsOptions;
+  }
+
+  private static Future<Void> updateHeaders(List<String> moduleIds, Map<String, String> headers) {
+    if (moduleIds.size() != 1) {
+      return Future.failedFuture(FAILED_TO_GET_USER_MODULE_ID);
+    } else {
+      headers.put(OKAPI_MODULE_ID_HEADER, moduleIds.get(0));
+      return Future.succeededFuture();
+    }
+  }
+
+  private static Future<JsonObject> requestCustomFieldsDefinitions(HttpClientInterface client, Map<String, String> headers) {
+    Future<JsonObject> future = Future.future();
+    try {
+      client.request(GET_CUSTOM_FIELDS_ENDPOINT, headers)
+        .whenComplete((response, ex) -> proceedResponse(future, response, ex, Response::getBody));
+    } catch (Exception e) {
+      future.fail(e);
+    }
+    return future;
+  }
+
+  private static Future<Void> updateCF(JsonObject cfCollection, Map<String, Set<String>> customFieldsOptions,
+                                       HttpClientInterface client, Map<String, String> okapiHeaders) {
+    Future<Void> future = Future.future();
+    boolean isUpdated = updateCfOptions(cfCollection, customFieldsOptions);
+    if (isUpdated) {
+      Map<String, String> headers =
+        createHeaders(okapiHeaders, HTTP_HEADER_VALUE_TEXT_PLAIN, HTTP_HEADER_VALUE_APPLICATION_JSON);
+      try {
+        client.request(HttpMethod.PUT, cfCollection, PUT_CUSTOM_FIELDS_ENDPOINT, headers)
+          .whenComplete((response, ex) -> proceedResponse(future, response, ex, r -> null));
+      } catch (Exception e) {
+        future.fail(e);
+      }
+    } else {
+      future.complete();
+    }
+    return future;
+  }
+
+  private static <T> void proceedResponse(Future<T> future, Response response, Throwable ex,
```

Comment:
This should probably be called "processResponse"?

Reply:
Yes. Fixed